common/telemetry/config.go

```diff
@@ -331,6 +331,7 @@ func (scse *sharedConnSpanExporter) Start(ctx context.Context) error {
 		var cc *grpc.ClientConn
 		cc, err = scse.dialer.Dial(ctx)
 		if err != nil {
+			// todo: check what return value is expected here
```

Comment:
Both of these are things that can be fixed long-term if we start using SBRM with fx. However, that change may not happen for a while. For the time being, let's just wrap these errors with pkg/errors and add the plumbing to propagate these errors to the Start method's frame and return them.